runtime/src/main/java/org/corfudb/runtime/view/LayoutBuilder.java

```diff
@@ -102,14 +111,98 @@ public LayoutWorkflowManager removeLayoutServers(Set<String> endpoints)
     }
 
     /**
+     * Adds a new layout server.
+     * @param endpoint Endpoint to be added.
+     * @return this.
+     */
+    public LayoutBuilder addLayoutServer(String endpoint) {
+        layout.getLayoutServers().add(endpoint);
+        return this;
+    }
+
+    /**
+     * Adds a new sequencer server.
+     * @param endpoint Endpoint to be added.
+     * @return this.
+     */
+    public LayoutBuilder addSequencerServer(String endpoint) {
+        layout.getSequencers().add(endpoint);
+        return this;
+    }
+
+    /**
+     * Adds a new logunit server.
+     * @param stripeIndex stripe index where new endpoint should be added.
+     * @param globalLogTail Global log tail to split segment.
+     * @param newLogunitEndpoint Endpoint to be added.
+     * @return this.
+     */
+    public LayoutBuilder addLogunitServer(int stripeIndex, long globalLogTail,
+                                          String newLogunitEndpoint) {
```

Comment:
I think more comments are required to explicate closed/open segments and split segments.